plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/query/HudiSnapshotDirectoryLister.java

```diff
@@ -44,10 +50,13 @@ public class HudiSnapshotDirectoryLister
     private final HudiTableHandle tableHandle;
     private final Lazy<HoodieTableFileSystemView> lazyFileSystemView;
     private final Lazy<Map<String, HudiPartitionInfo>> lazyAllPartitionInfoMap;
+    private final Optional<HudiIndexSupport> lazyIndexSupportOpt;
 
     public HudiSnapshotDirectoryLister(
+            ConnectorSession session,
             HudiTableHandle tableHandle,
             boolean enableMetadataTable,
+            Lazy<HoodieTableMetadata> lazyTableMetadata,
```

Comment:
`lazyTableMetadata` could be used to construct `lazyFileSystemView` so the table metadata is reused.

Reply:
Addressed.